### DISCUSSION TITLE
Cert error for Azure DSN (single and SAN)

### DISCUSSION THREAD

**Discussion by singleton2787** [rcl-ssl/docs#45] [General]
How to I get the txt record correct?

Error:Could not create Azure DNS certificate : impactaiinc.com . Could not validate order https://acme-v02.api.letsencrypt.org/acme/order/1707120397/266299283747 with challenge DNS, status reported : invalid. DNS problem: NXDOMAIN looking up TXT for _acme-challenge.impactaiinc.com - check that a DNS record exists for this domain Did not find the DNS TXT record with your DNS provider and/or the DNS TXT record has the wrong name or value. ERR0

**Comment by singleton2787** [comment]
I had txt record but it didn't like the format....

**Comment by rcladmin** [comment]
Give me a little time I am looking into your issue.

**Comment by rcladmin** [comment]
I am working on this theory. There is an old TXT record at _acme-challenge.impactaiinc.com : "swJXb6fNNt86-9ZgYzubJbl4WyMWhkB9JZZVnqG2mTM". This can be from a previous validation for a previous Let Encrypt Certificate. For the new validation request, Azure DNS may be blocking the creation of the new DNS TXT record, possibly because of some security configuration on your end. The Let's encrypt server finds the old record and the validation fails.

To find a possible solution, check to see if there is some fine grained security setting preventing your Microsoft AD Account from adding DNS Text Records to Azure DNS. You can check your RBAC in IAM for your Azure DNS Zone. Please consult this link.

https://learn.microsoft.com/en-us/azure/dns/dns-protect-zones-recordsets

To create a DNS Record your RBAC should be set to contributor will full access to Add/edit/Delete DNS TXT records. 

You can also set contributor role on your subscripion by following this:

https://docs.rclapp.com/authorization/access-control-user.html

Yu can also check to see if your resource locks is set to Read Only for your Azure DNS Zone.

I am looking forward to hearing about your findings on this.

**Comment by rcladmin** [comment]
- Login to the Azure Portal with the same Microsoft AD Credentials  you use to login to the RCL SSL portal
- Go to your Azure DNS Zone and in IAM and click on the "View my access" button
- Your role should be DNS Zone Contributor
- Try this and see what you get, keep me posted on your findings

![zone-rbac](https://github.com/rcl-ssl/docs/assets/4339259/a95a72e6-d044-4860-96b2-11ae4f8b61c4)

**Comment by singleton2787** [comment]
The record in question is hosted outside Azure in GreenGeeks. How do I do that? Do old certs need the existing txt record to renew? Mike SingletonCTOImpact AI Inc 704.322.2015Sent from my iPhoneOn May 3, 2024, at 5:05 PM, RCL ***@***.***> wrote:﻿
I am working on this theory. There is an old TXT record at _acme-challenge.impactaiinc.com : "swJXb6fNNt86-9ZgYzubJbl4WyMWhkB9JZZVnqG2mTM". This can be from a previous validation for a previous Let Encrypt Certificate. For the new validation request, Azure DNS may be blocking the creation of the new DNS TXT record, possibly because of some security configuration on your end. The Let's encrypt server finds the old record and the validation fails.
To find a possible solution, check to see if there is some fine grained security setting preventing your Microsoft AD Account from adding DNS Text Records to Azure DNS. You can check your RBAC in IAM for your Azure DNS Zone. Please consult this link.
https://learn.microsoft.com/en-us/azure/dns/dns-protect-zones-recordsets
To create a DNS Record your RBAC should be set to contributor will full access to Add/edit/Delete DNS TXT records. Could you look into this. I am looking forward to hearing about your findings on this.

**Comment by rcladmin** [comment]
You can use the **Stand Alone** or **Stand Alone SAN** option, to create a certificate in the RCL SSL Portal if your DNS Records are hosted outside Azure (GreenGeeks).

- If you can manually add a DNS TXT Record in your DNS management portal (in cPanel), use the **DNS Challenge** in the stand alone option
- If you can place a file in the root of your website, use the **HTTP Challenge** in the stand alone option

https://docs.rclapp.com/portal/stand-alone.html
https://docs.rclapp.com/portal/stand-alone-san.html

If you want to use the **Azure DNS** option in the RCL SSL Portal with an Azure DNS Zone, you need to **delegate** you DNS from your dns provider to an Azure DNS Zone

https://docs.rclapp.com/portal/azure-dns.html#create-a-dns-zone-and-configure-name-server

Old certificates DO NOT need the existing record to renew. What you need to do is create a new certificate and that new certificate will use a new DNS TXT record to validate. Once validated, the certificate will be created. You then need to delete old certificate and replace it with the new certificate in your web server. I don't know how GreenGeeks works with SSL certificates, whether the entire certificate creation/renewal process is a manual or an automatic process.
